src/Training.AirBnb.Clone.Backend/Backend Project/Domain/Services/CountryService.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Backend_Project.Domain.Entities;
+using Backend_Project.Domain.Exceptions.CountryExceptions;
+using Backend_Project.Domain.Interfaces;
+using Backend_Project.Persistance.DataContexts;
+using System.Linq.Expressions;
+
+namespace Backend_Project.Domain.Services
+{
+    public class CountryService : IEntityBaseService<Country>
+    {
+        private IDataContext _appDataContext;
+        public CountryService(IDataContext dataContext)
+        {
+            _appDataContext = dataContext;
+        }
+
+        public async ValueTask<Country> CreateAsync(Country country, bool saveChanges = true, CancellationToken cancellationToken = default)
+        {
+            if (GetUndeletedCountries().Any(c => c.Name.Equals(country.Name)))
+                throw new CountryAlreadyExistsException("This Country already Exists");
```

Comment:
Bu joydagi validatsiyani alohida methodga ko'chirish kerak. Single responsibilityga to'g'ri kelmay qoladi.